src/plugins/intel_cpu/src/nodes/fullyconnected.cpp

```diff
@@ -1116,6 +1108,80 @@ bool FullyConnected::tryExtractParamForLLMFc(llmdnn::fc_create_param& param) {
     return false;
 }
 
+MemoryPtr FullyConnected::castMemoryPtr(MemoryPtr memPtr, const InferenceEngine::Precision prec) {
+    auto engine = memPtr->getEngine();
+    dnnl::stream engine_stream(engine);
+
+    // get weightPtr and dnnlweightDesc
+    auto dnnlMemDesc = memPtr->GetDescWithType<DnnlMemoryDesc>()->getDnnlDesc();
+    // desc_ptr
+    auto newMemDescPtr = memPtr->getDescPtr()->cloneWithNewPrecision(prec);
+    // dnnl_desc_ptr
+    DnnlMemoryDescPtr dnnlNewMemPtr = MemoryDescUtils::convertToDnnlMemoryDesc(newMemDescPtr);
+    // dnnl_desc
+    auto dnnlNewMemDesc = dnnlNewMemPtr->getDnnlDesc();
+    MemoryPtr newMemPtr = MemoryPtr(new Memory(engine));
+    newMemPtr->Create(dnnlNewMemPtr);
+
+    auto reorder = getReorderPrim(context->getParamsCache(), engine, dnnlMemDesc, dnnlNewMemDesc);
+    primArgs = {{DNNL_ARG_SRC, memPtr->GetPrimitive()}, {DNNL_ARG_DST, newMemPtr->GetPrimitive()}};
+    reorder.execute(engine_stream, primArgs);
+    engine_stream.wait();
+    return newMemPtr;
+}
+
+void FullyConnected::tryCompressWeightForLLMFc() {
```

Comment:
we'd better to refactor to extract the common part into a function from tryCompressWeightForLLMFc and tryUseLLMFc.

Reply:
@luo-cheng2021 refactor here.